src/base/utils/fs.cpp

```diff
@@ -313,12 +325,20 @@ bool Utils::Fs::isNetworkFileSystem(const QString &path)
     if (statfs(file.toLocal8Bit().constData(), &buf) != 0)
         return false;
 
-#if defined(Q_OS_MAC) || defined(Q_OS_OPENBSD)
     // XXX: should we make sure HAVE_STRUCT_FSSTAT_F_FSTYPENAME is defined?
     return ((strncmp(buf.f_fstypename, "cifs", sizeof(buf.f_fstypename)) == 0)
         || (strncmp(buf.f_fstypename, "nfs", sizeof(buf.f_fstypename)) == 0)
         || (strncmp(buf.f_fstypename, "smbfs", sizeof(buf.f_fstypename)) == 0));
 #else
+    QString file = path;
```

Comment:
Why did you prefer to duplicate common part of code? Why not:
```c++
#if defined(Q_OS_WIN)
// Windows code
#else
// Common *nix code
#if defined(Q_OS_MAC) || defined(Q_OS_OPENBSD)
// BSD code
#else
// Linux code
#endif
#endif
```

Reply:
Mostly because I easily get lost in nested #if conditions, so having only 1 level of #if is much more readable for me.
Also `struct statfs buf;` in BSD & Linux is used quite differently, IMO it is OK to treat them as independent code (by having no common code section).